scripts/check_license_headers.sh

```diff
@@ -0,0 +1,11 @@
+set -e
+echo "Checking that a Apache 2 license header is present at the top of all scala files in src/"
+
+src_files_with_missing_license=$(find src -type f -name '*.scala' -exec bash -c '[ ! -z "$(head -1 {} | grep -v "// Licensed to the Apache Software Foundation (ASF) under one")" ] && echo "{}"' \;)
+if [ ! -z "$src_files_with_missing_license" ]
```

Comment:
`[ ! -z "${something}"]` is the same as `[ -n "${something}" ]`. Furthermore, `-n` is the "default test", so this can be further simplified to just `[ "${something}" ]`, which is idiomatic Bash.